src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -88,6 +88,17 @@ public class FhirCosmosSearchService : SearchService
                 cancellationToken);
         }
 
+        protected override async Task<SearchResult> SearchForReindexInternalAsync(
+            SearchOptions searchOptions,
+            string searchParameterHash,
```

Comment:
nit: do we need param validation for searchParameterHash?

Reply:
Down the call chain, I have updated the QueryBuilder.cs to check for null.